matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/actions/MegolmSessionDataImporter.kt

```diff
@@ -76,7 +77,11 @@ internal class MegolmSessionDataImporter @Inject constructor(private val olmDevi
                     outgoingGossipingRequestManager.cancelRoomKeyRequest(roomKeyRequestBody)
 
                     // Have another go at decrypting events sent with this session
-                    decrypting.onNewSession(megolmSessionData.senderKey!!, sessionId!!)
+                    when (decrypting) {
```

Comment:
For things like that I used the extension pattern. For example there is the IMXWithHeldExtension that is only supported by MXMegolmDecryption and not by MXOlmDecryption. So in code you can do 
```
if (alg is IMXWithHeldExtension) {
            alg.onRoomKeyWithHeldEvent(withHeldContent)
        }
```

Maybe we can use also this principles for new session? Or let me know if you think it's too overengineered

Reply:
I think that could be quite nice as we're not using any of the other functions from the `MXMegolmDecryption`

maybe a bit of a _hot take_ but I'm also wondering if we need the base interface? it seems that only `fun decryptEvent(event: Event, timeline: String): MXEventDecryptionResult` is used by both olm and megolm

when talking to the `roomDecryptorProvider.getOrCreateRoomDecryptor` we tend to already know which type of decrypter we need, in the case of `MegolmSessionDataImporter` I wouldn't expect to see any olm decrypters popping up